src/Features/Core/Portable/Completion/CompletionServiceWithProviders.ProviderManager.cs

```diff
@@ -116,33 +117,22 @@ public ImmutableArray<CompletionProvider> GetProviders(ImmutableHashSet<string>?
                 return allCompletionProviders.ConcatFast(projectCompletionProviders);
             }
 
-            private ImmutableArray<CompletionProvider> GetProjectCompletionProviders(Project? project)
+            public ImmutableArray<CompletionProvider> GetProjectCompletionProviders(Project? project)
```

Comment:
why not make this async.  i'm not sure the need for this to be synchronous.

Reply:
Yes, sure I can do that. It's just that this is called by a synchronous public API of ours (which is in turn called synchronously by Editor), so it will just move the synchronousness up instead.

Reply:
which synchronous API calls this?

Reply:
It's called in [this](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionServiceWithProviders.cs,237) and [this](https://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/IntelliSense/AsyncCompletion/CompletionSource.cs,182)